### ISSUE TITLE
disallow importing macros: defineProps etc

### ISSUE THREAD

**Issue by dcecile** [vuejs/eslint-plugin-vue#2437]
It it worth tracking approved but closed #1906 as a new issue here?

Example playground that includes [vue/prefer-import-from-vue rule](https://eslint.vuejs.org/rules/prefer-import-from-vue.html) enabled: [link](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNzsEKgzAMxvFXCTltsOG9dGWPsAfIRTRKQdPSRmGUvvsq7uAx/D/4peAQRkaDVnmNS6/sSADs6HdXCkwhQK22O04S211GNg/JR4XMukVYeplfhJoJW/NrDEmhwMiTF/6kEDNUmFJYgXDfmJDk0mw50IYZyJq8zCTV3e6HeCoOHxj7lDmhKU1pD7/1G/msT86LF+3+i1p/8rNMrg==)

This playground code triggers a compiler message about the macro import:

```
[@vue/compiler-sfc] `defineProps` is a compiler macro and no longer needs to be imported.
```

> **Please describe what the rule should do:** It should warn if defineProps, defineEmits and similar are imported from vue (autofixable)
> 
> **What category should the rule belong to?**
> 
>   [x] Enforces code style (layout)
>   [ ] Warns about a potential error (problem)
>   [ ] Suggests an alternate way of doing something (suggestion)
>   [ ] Other (please specify:)
> 
> **Provide 2-3 code examples that this rule should warn about:**
> 
> bad:
> 
> ```
> <script setup>
> import { defineProps } from '@vue/runtime-dom';
> 
> defineProps<{}>();
> </script>
> ```
> 
> good:
> 
> ```
> <script setup>
> defineProps<{}>();
> </script>
> ```
> 
> good:
> 
> ```
> <script setup>
> import { defineProps } from 'my-lib';
> 
> defineProps('whatever');
> ```
> 
> **Additional context** Useful when `define*` are added to eslint globals, there's no need to import them manually. If project was using imports and then dev switched to globals then autofix would just automatically remove unnecessary imports

**Comment by scandar** [comment]
I found a workaround until this is officially implemented 

You could use [no-restricted-imports](https://eslint.org/docs/latest/rules/no-restricted-imports) to restrict certain imports.

```ts
"no-restricted-imports": [
  "error",
  {
    paths: [
      {
        name: "vue",
        importNames: ["defineProps", "defineEmits", "defineExpose"],
        message: "Compiler macros no longer need to be imported. Please remove the import.",
      },
    ],
  },
]
```

**Comment by yoyo837** [comment]
> "no-restricted-imports": [
>   "error",
>   {
>     paths: [
>       {
>         name: "vue",
>         importNames: ["defineProps", "defineEmits", "defineExpose"],
>         message: "Compiler macros no longer need to be imported. Please remove the import.",
>       },
>     ],
>   },
> ]

How about if `eslint-plugin-vue` for `vue@3` had this rule built in?
